core/src/main/scala/com/databricks/labs/remorph/intermediate/trees.scala

```diff
@@ -10,36 +12,34 @@ import scala.util.control.NonFatal
 private class MutableInt(var i: Int)
 
 case class Origin(
-    line: Option[Int] = None,
-    startPosition: Option[Int] = None,
-    startIndex: Option[Int] = None,
-    stopIndex: Option[Int] = None,
-    sqlText: Option[String] = None,
-    objectType: Option[String] = None,
-    objectName: Option[String] = None)
-
-object CurrentOrigin {
-  private val value = new ThreadLocal[Origin]() {
-    override def initialValue: Origin = Origin()
-  }
+    startLine: Option[Int] = None,
+    startColumn: Option[Int] = None,
+    endLine: Option[Int] = None,
+    endColumn: Option[Int] = None,
+    startTokenIndex: Option[Int] = None,
+    endTokenIndex: Option[Int] = None)
```

Comment:
Is it possible/meaningful for some of the fields to be `Some` while others are `None`, or is it either all fields `None` or all fields `Some` ?

In the latter case, we would be better to either `origin` be an `Option[Origin]` or, even better, define `Origin` as
```scala
sealed trait Origin
case object UnknownOrigin extends Origin
case class InputOrigin(
    startLine: Int,
    startColumn: Int,
    endLine: Int,
    endColumn: Int,
    startTokenIndex: Int,
    endTokenIndex: Int) extends Origin

// optionally, adding the following case for the node that are synthesized during optimization phase
// could be handy.
case class SyntheticOrigin(synthesizedBy: String) extends Origin
```